### PR TITLE
Cluster resource classic unit tests

### PR DESCRIPTION
**Pull request by isaacdorfman**

### CODE REVIEW

File: provider/cluster_resource_classic_test.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	. "github.com/onsi/ginkgo/v2/dsl/core" // nolint
+	. "github.com/onsi/gomega"             // nolint
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+)
+
+var _ = Describe("Cluster creation", func() {
```
Comment (bardielle) [inline provider/cluster_resource_classic_test.go:31]: *ROSA*

File: provider/cluster_resource_classic_test.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	It("Creates rosa classic cluster", func() {
+		clusterState := &ClusterRosaClassicState{
+			Name: types.String{
```
Comment (bardielle) [inline provider/cluster_resource_classic_test.go:34]: Please use a global variable for a common variables

File: provider/cluster_resource_classic_test.go (hunk 1 above)
Comment (bardielle) [inline provider/cluster_resource_classic_test.go:31]: use beforeach for a common setup
and aftereach

File: provider/cluster_resource_classic_test.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	. "github.com/onsi/ginkgo/v2/dsl/core" // nolint
+	. "github.com/onsi/gomega"             // nolint
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+)
+
+var _ = Describe("Cluster creation", func() {
+	It("Creates rosa classic cluster", func() {
```
Comment (bardielle) [inline provider/cluster_resource_classic_test.go:32]: Describe with more details about the test

File: provider/cluster_rosa_classic_resource.go [hunk 4]
```diff
@@ -689,7 +701,6 @@ func (r *ClusterRosaClassicResource) populateState(ctx context.Context, object *
 
 		thumbprint, err := getThumbprint(sts.OIDCEndpointURL())
 		if err != nil {
-			r.logger.Error(ctx, "cannot get thumbprint", err)
```
Comment (bardielle) [inline provider/cluster_rosa_classic_resource.go:692]: why did removed the log?

File: provider/cluster_resource_classic_test.go (hunk 1 above)
Comment (isaacdorfman) [reply provider/cluster_resource_classic_test.go]: Done

File: provider/cluster_resource_classic_test.go (hunk 2 above)
Comment (isaacdorfman) [reply provider/cluster_resource_classic_test.go:51]: Done

File: provider/cluster_resource_classic_test.go (hunk 3 above)
Comment (isaacdorfman) [reply provider/cluster_resource_classic_test.go]: Done

File: provider/cluster_rosa_classic_resource.go (hunk 4 above)
Comment (isaacdorfman) [reply provider/cluster_rosa_classic_resource.go:692]: Done

File: provider/cluster_resource_classic_test.go [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	. "github.com/onsi/ginkgo/v2/dsl/core" // nolint
+	. "github.com/onsi/gomega"             // nolint
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/openshift-online/ocm-sdk-go/logging"
+)
+
+var _ = Describe("Rosa Classic cluster creation", func() {
+	clusterState := &ClusterRosaClassicState{}
```
Comment (bardielle) [inline provider/cluster_resource_classic_test.go:33]: wrap it with 
`
var (
	clusterState := &ClusterRosaClassicState{}
	clusterId := "1n2j3k4l5m6n7o8p9q0r"
	clusterName := "my-cluster"
	regionId := "us-east-1"
	multiAz := true
	rosaCreatorArn := "arn:aws:iam::123456789012:dummy/dummy"
	apiUrl := "https://api.my-cluster.com:6443"
	consoleUrl := "https://console.my-cluster.com"
	machineType := "m5.xlarge"
	availabilityZone := "us-east-1a"
	ccsEnabled := true
	awsAccountID := "123456789012"
	privateLink := false
	oidcEndpointUrl := "example.com"
	roleArn := "arn:aws:iam::123456789012:role/role-name"

)
`

**Review by bardielle — COMMENTED** [review]
/lgtm

**Review by nirarg — COMMENTED** [review]
For both functions you test, you perform only one test - the "happy flow", aren't there any other scenarios to verify?
For example, in `createClassicClusterObject`, shouldn't you check that in case there are multiple `AvailabilityZones` that the code append those in the right way?

**Comment by isaacdorfman** [comment]
@nirarg 

> For both functions you test, you perform only one test - the "happy flow", aren't there any other scenarios to verify? For example, in `createClassicClusterObject`, shouldn't you check that in case there are multiple `AvailabilityZones` that the code append those in the right way?

I added this check now.

File: provider/cluster_resource_classic_test.go (hunk 5 above)
Comment (isaacdorfman) [reply provider/cluster_resource_classic_test.go]: Done

File: provider/cluster_resource_classic_test.go [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+			"private_link": privateLink,
+			"sts": map[string]interface{}{
+				"oidc_endpoint_url": oidcEndpointUrl,
+				"role_arn":          roleArn,
+			},
+		},
+	}
+}
+
+var _ = Describe("Rosa Classic cluster creation", func() {
```
Comment (nirarg) [inline provider/cluster_resource_classic_test.go:111]: This is worng description, `populateRosaClassicClusterState` is not related only to the creation, it used by all crud functions

File: provider/cluster_rosa_classic_resource.go [hunk 7]
```diff
@@ -787,10 +837,12 @@ func (r *ClusterRosaClassicResource) populateState(ctx context.Context, object *
 			Null: true,
 		}
 	}
-	version, ok := object.Version().GetID()
+
+	version := object.Version()
```
Comment (bardielle) [inline provider/cluster_rosa_classic_resource.go:841]: please back to use it as an version

File: provider/cluster_rosa_classic_resource.go [hunk 8]
```diff
@@ -600,32 +626,48 @@ func (r *ClusterRosaClassicResource) populateState(ctx context.Context, object *
 	state.Name = types.String{
 		Value: object.Name(),
 	}
+
+	// region := object.Region()
```
Comment (bardielle) [inline provider/cluster_rosa_classic_resource.go:630]: please remove all the comment lines

**Review by nirarg — APPROVED** [review]

**Pull request merged**